vagrant/provisioning/roles/arkcase-app/tasks/main.yml

```diff
@@ -697,19 +704,32 @@
   become_user: arkcase
   template:
     backup: yes
-    src: server-v3.3.2.yaml
+    src: server.yml
     dest: /home/arkcase/.arkcase/acm/acm-config-server-repo/arkcase-server.yaml
-  when: arkcase_version != "" and arkcase_version is version('3.3.2', '>=') and arkcase_version is version('3.3.3', '<')
-    
-# this template includes encrypted passwords
-# so it must be run after we encrypt the passwords
-- name: server-specific configuration
+
+# the rolesToGroups format changed during the 3.3.3 release cycle, so we need 
+# one format for pre-3.3.3 versions, and another format for 3.3.3+
+- name: roles to groups mapping (3.3.2 and below)
   become: yes
   become_user: arkcase
-  template:
-    backup: yes
-    src: server.yml
-    dest: /home/arkcase/.arkcase/acm/acm-config-server-repo/arkcase-server.yaml
+  blockinfile:
+    path: /home/arkcase/.arkcase/acm/acm-config-server-repo/arkcase-server.yaml
+    marker: "# {mark} ANSIBLE MANAGED BLOCK - ROLES TO GROUPS"
+    block: |
+      application.roles_to_groups: "{{ roles_to_groups }}"
+  when: arkcase_version != "" and arkcase_version is version('3.3.3', '<')
+
+# the rolesToGroups format changed during the 3.3.3 release cycle, so we need 
```

Comment:
pulled rolesToGroups out of the template completely, and added a conditional blockinfile to add the right format.